src/waypoints/route.py

```diff
@@ -1,160 +1,150 @@
+def set_start(coords, map_widget, start_marker, end_marker):
+    """Set the widget start indicator at the coords."""
     print("Start set to ", coords)
     start_marker.set_text("Start")
     start_marker.set_position(coords[0], coords[1])
 
     map_widget.delete_all_polygon()
-    generate_box(start_marker.position, end_marker.position)
+    generate_box(start_marker.position, end_marker.position, map_widget)
 
 
-def set_end(coords):
+def set_end(coords, map_widget, start_marker, end_marker):
+    """Set the widget end indicator at the coords."""
     print("End set to ", coords)
     end_marker.set_text("End")
     end_marker.set_position(coords[0], coords[1])
 
     map_widget.delete_all_polygon()
-    generate_box(start_marker.position, end_marker.position)
+    generate_box(start_marker.position, end_marker.position, map_widget)
 
 
 def left_click_event(coordinates_tuple):
+    """Left click event handler."""
     print("Left click event with coordinates:", coordinates_tuple)
 
 
-def bind_events(map_widget):
-
+def bind_events(map_widget, start_marker, end_marker):
+    """Bind click events to the map widget."""
     map_widget.add_left_click_map_command(left_click_event)
-
-    map_widget.add_right_click_menu_command(label="Set start",
-                                            command=set_start,
-                                            pass_coords=True)
-
-    map_widget.add_right_click_menu_command(label="Set end",
-                                            command=set_end,
-                                            pass_coords=True)
+    map_widget.add_right_click_menu_command(
+        label="Set start",
+        command=lambda coords: set_start(coords, map_widget, start_marker, end_marker),
+        pass_coords=True,
+    )
+    map_widget.add_right_click_menu_command(
+        label="Set end",
+        command=lambda coords: set_end(coords, map_widget, start_marker, end_marker),
+        pass_coords=True,
+    )
 
 
 def yes_or_no(question):
-    while "the answer is invalid":
+    """Yes or no helper function."""
+    while True:
         reply = str(input(question + ' (y/n): ')).lower().strip()
         if reply[0] == 'y':
             return True
         if reply[0] == 'n':
             return False
 
 
-def fetch_data(pointA, pointB):
-
-    # Get the final bounding box
-    box = bounding_box(pointA[0],
-                       pointA[1],
-                       pointB[0],
-                       pointB[1],
-                       box_buffer)
+def fetch_data(point_a, point_b):
+    """Fetch map data for two points."""
+    box = bounding_box(point_a[0], point_a[1], point_b[0], point_b[1], BOX_BUFFER)
 
-    if (box_size(box) > 0.25):
+    if box_size(box) > 0.25:
         print("Your bounding box is too large! (over 25 square degrees)")
-        exit()
+        sys.exit()
 
     box_string = f"{box[0]},{box[1]},{box[2]},{box[3]}"
-
-    # FIXME: this may generate edges that are split (nodes outside bounding box
```

Comment:
Just wondering, was this fixed or do you just think that it's a minor enough issue that we don't need to have a comment about it anymore?